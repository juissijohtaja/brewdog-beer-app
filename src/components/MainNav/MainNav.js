import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import EqualizerIcon from '@material-ui/icons/Equalizer'
import LocalDrinkIcon from '@material-ui/icons/LocalDrink'
import Box from '@material-ui/core/Box'

const useStyles = makeStyles((theme) => ({
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  navTheme: {
    backgroundColor: theme.palette.primary.main
  },
}))

const MainNav = () => {
  const classes = useStyles()

  return (
    <Box flexGrow={1} data-testid='main-nav'>
      <AppBar position='static' className={classes.navTheme} aria-label='main menu'>
        <Toolbar>
          <LocalDrinkIcon className={classes.menuButton} />
          <Typography variant='h6' className={classes.title}>
              Brewdog Beers {'&'} Food Matcher
          </Typography>
        </Toolbar>
      </AppBar>
    </Box>
  )
}

export default MainNav