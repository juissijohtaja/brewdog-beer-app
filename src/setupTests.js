import '@testing-library/jest-dom'

const initialStoreState = {
  beers: [],
  keyword: '',
  pagination: {
    page: 0,
    rowsPerPage: 10
  },
  sortDesc: true
}

export const changedStoreState = {
  beers: [{
    id: 999,
    beer: 1111,
    description: 'Lorem ipsum dolor sit amet',
    startDate: '1999-12-31T00:00:00+00:00',
    category: 'Test',
    responsible: 'John Doe',
    savingsAmount: 1234.56,
    currency: 'EUR',
    complexity: 'Simple'
  },
  {
    id: 9998,
    beer: '',
    description: '',
    startDate: '',
    category: '',
    responsible: '',
    savingsAmount: '',
    currency: '',
    complexity: ''
  }],
  keyword: 'lorem',
  pagination: {
    page: 0,
    rowsPerPage: -1
  },
  sortDesc: false
}

export default initialStoreState