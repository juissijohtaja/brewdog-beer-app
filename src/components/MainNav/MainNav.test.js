import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { initialStoreState } from '../../setupTests'
import MainNav from './MainNav'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

describe('MainNav Component', () => {
  let store, component
  const mockStore = configureStore([])

  beforeEach(() => {
    store = mockStore(initialStoreState)
    store.dispatch = jest.fn()

    component = renderer.create(
      <Provider store={store}>
        <MainNav />
      </Provider>
    )
  })

  it('MainNav should render AppBar, EqualizerIcon and Beers', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
