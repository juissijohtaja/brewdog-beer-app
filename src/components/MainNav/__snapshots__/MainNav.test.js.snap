// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav Component MainNav should render AppBar, EqualizerIcon and Beers 1`] = `
<div
  className="MuiBox-root MuiBox-root-4"
  data-testid="main-nav"
>
  <header
    aria-label="main menu"
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-navTheme-3 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-menuButton-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z"
        />
      </svg>
      <h6
        className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
      >
        Beers
      </h6>
    </div>
  </header>
</div>
`;
